src/net.cpp

```diff
@@ -1053,6 +1053,8 @@ bool CConnman::AddConnection(const std::string& address, ConnectionType conn_typ
     case ConnectionType::BLOCK_RELAY:
         max_connections = m_max_outbound_block_relay;
         break;
+    case ConnectionType::FULLRBF:
+        max_connections = m_max_outbound_fullrbf_relay;
     // no limit for ADDR_FETCH because -seednode has no limit either
```

Comment:
This is getting out of scope for this review. But as an aside, IIUC this function includes manually added connections in its max connections count. I'm not sure that's correct. If I manually add a bunch of nodes that match some category, that doesn't necessarily mean I _only_ want my node connecting to the manually added nodes.

Reply:
Oh, actually, looks like this code doesn't do anything in practice, as `AddConnection` is only used by the `addconnection` RPC call.

Reply:
Oh right, that's correct my mistake. IMHO, we do have a confusing connection type with a "flat" approach instead of an attribute based one as argued previously in https://github.com/bitcoin/bitcoin/pull/18044#pullrequestreview-442115002. I'll fix that soon (hiking!)

Reply:
As you raised, manually adding a bunch of nodes shouldn't necessarily mean that you only want your node to connect to manually added nodes, as you might like to profit from the automatic discovery and connecting logic. It sounds really artificial for us to constraint node operators in their peers topology when there is a deliberate choice to deviate from the spontaneous topology generated by `ThreadOpenConnections`.

That said, this approach of enforcing connection limits controls on the types of connection is already what is implemented by `AddConnection`, not subjecting outbound full-rbf peers to limits control would deviate from that behavior. So I think the real discussion we should have would be about if those checks make really sense in `ThreadOpenConnections`. Though as it's a point preceding this PR, I would leave as it is.

Beyond, I wonder if that really matters as `addconnection` is regtest-only (cf. L347, in `src/rpc/net.cpp`).